Sources/AWSLambdaRuntime/LambdaRuntimeClient.swift

```diff
@@ -188,6 +197,10 @@ final actor LambdaRuntimeClient: LambdaRuntimeClientProtocol {
         }
     }
 
+    private func writeCustomHeader(_ buffer: NIOCore.ByteBuffer) async throws {
+        _hasStreamingCustomHeaders = true
```

Comment:
Any reason you don't pass `_hasStreamingCustomHeaders` as a parameter through all the functions.
Otherwise if you are going to add some state perhaps you should add it to the `LambdaRuntimeClient.state`.

Reply:
That's a good point.  The `_hasStreamingCustomHeaders` is indeed a state of the current response (not the connection, as it can be reused for multiple next() and their responses)

I'll check if I can move this flag to one of the many states struct present in the `LambdaRuntimeClient` :-)

Reply:
@adam-fowler
Regardless or where I store this value (in an existing state, if appropriate, or in a Bool) ...
What is best in terms of API? Adding that flag (with a default value) to all `write()` methods on the protocol ? Or add a new fun to the protocol ?

Reply:
@adam-fowler after having reviewed the existing code, I conclude that there is an easier way to pass that flag down to the channel handler.  Look at the new version. Thank you for having challenged me on this!